Computer History Museum Archive.js

```diff
@@ -0,0 +1,1000 @@
+	const fields = [...doc.querySelectorAll("div.field")];
+
+	let newItem = new Zotero.Item("book");
+	newItem.archive = 'Computer History Museum';
+
+	fields.forEach((nodeOriginal) => {
+		// clone the node to avoid altering the document.
+		const node = nodeOriginal.cloneNode(true)
+		const fieldTitleNode = node.querySelector("h4")
+		const fieldTitle = fieldTitleNode.textContent.trim()
+		switch(fieldTitle) {
+			case 'Title':
+				newItem.title = getContent(node, fieldTitle)
+				break;
+			case 'Date':
+				newItem.date = getContent(node, fieldTitle)
+				break;
+			case 'Participants':
+			case 'Contributor':
+				newItem.creators = getContributors(node)
+				break;
+			case 'Publisher':
+				newItem.publisher = getContent(node, fieldTitle)
+				break;
+			case 'Place of Publication':
+				newItem.publisher += '; ' + getContent(node, fieldTitle)
+				break;
+			case 'Extent':
+				newItem.numPages = getContent(node, fieldTitle).replace(' p.', '')
+				break;
+			case 'Lot Number':
+				newItem.archiveLocation = getContent(node, fieldTitle)
+				break;
+			case 'Description':
+				newItem.abstractNote = insertToTheStartOfAbstract(getContent(node, fieldTitle), newItem.abstractNote)
+				break;
+			case 'Biographical Notes':
+				newItem.abstractNote = insertToTheEndOfAbstract(getContent(node, fieldTitle), newItem.abstractNote, fieldTitle)
+				break
+			case 'Copyright Holder':
+				newItem.rights = getContent(node, fieldTitle)
+				break;
+			// the categories vary therefore many are collapsed under 'extra'.
+			case 'Type':
+			case 'Subject':
+			case 'Category':
+			case 'Collection Title':
+			case 'Copyright Holder':
+			case 'Dimensions':
+			case 'Credit':
+			case 'Place Manufactured':
+			case 'Manufacturer':
+			case 'System Requirements':
+			case 'Format':
+			case 'Series Title':
+			case 'Duration':
+			case 'Platform':
+			case 'Catalog Number':
+				newItem.extra = addToExtra(newItem.extra, getContent(node, fieldTitle), fieldTitle);
+				break
+			// parsing identifying numbers with existing Zotero fields e.g. callNumber & ISBN
+			case 'Identifying Numbers':
+				const numberDivs = node.querySelectorAll('tr')
+				numberDivs.forEach(div => {
+					const numberKey = div.querySelector('td.col1').textContent
+					const numberVal = div.querySelector('td.col2').textContent
+					switch(numberKey) {
+						case 'ISBN10':
+							newItem.ISBN = numberVal
+							break;
+						case 'LOC call num':
+							newItem.callNumber = numberVal
+							break;
+						default:
+							newItem.extra = addToExtra(newItem.extra, numberVal, numberKey)
+							break
+					}
+
+				})
+				
+
+		}
+
+	});
+
+	// add pdf documents
+	let pdfPath = undefined;
+	const pdfDiv = doc.querySelectorAll('div.mediaDocument li a');
+	if (pdfDiv) {
+		pdfDiv.forEach((div, pdfIndex) => {
+			const fileName = div.textContent;
+			pdfPath = div.href
+			if (pdfPath) {
+				newItem.attachments.push({
+					url: pdfPath.replace('https', 'http'),
```

Comment:
why get rid of https?

Reply:
For some reasons, the files provided by CHM can only be fetched by Zotero with http.

Reply:
All the files in your tests appear to be accessible via HTTPS, and are linked that way from the pages. If you can't access them that way, that's a problem with your internet connection.

Reply:
The files download fine to Zotero for me over HTTPS. This will need to be removed. We can't accept a translator that converts HTTPS URLs into HTTP ones.

If you're having trouble with your Zotero downloading files, we can debug that separately in the Zotero Forums. (It's likely some sort of issue with your internet connection, a proxy server, or security software on your computer.)

Reply:
Agreed. It's odd that it doesn't automatically redirect to HTTPS, for that matter.

Reply:
Thanks @dstillman and @AbeJellinek ! I’ve removed the conversions. Please review or advise for further improvement.